### ISSUE TITLE
Strange behavior with node title when zooming out 

### ISSUE THREAD

**Issue by SadE54** [thedmd/imgui-node-editor#128]
I'm  using the blueprint example as a base . Recently I used the snippets provided in widgets example to be able to add a combo like system : 
```
ed::Suspend();
            // There is some stately stuff happening here.  You call "open popup" exactly once, and this
            // causes it to stick open for many frames until the user makes a selection in the popup, or clicks off to dismiss.
            // More importantly, this is done inside Suspend(), so it loads the popup with the correct screen coordinates!
            if (do_popup) {
                ImGui::OpenPopup("popup_button"); // Cause openpopup to stick open.
                do_popup = false; // disable bool so that if we click off the popup, it doesn't open the next frame.
            }
            // This is the actual popup Gui drawing section.
            if (ImGui::BeginPopup("popup_button")) {
                // Note: if it weren't for the child window, we would have to PushItemWidth() here to avoid a crash!
                ImGui::TextDisabled("Pick One:");
                ImGui::BeginChild("popup_scroller", ImVec2(100, 100), true, ImGuiWindowFlags_AlwaysVerticalScrollbar);
                if (ImGui::Button("Option 1")) {
                    portable_strcpy(popup_text, "Option 1");
                    ImGui::CloseCurrentPopup();  // These calls revoke the popup open state, which was set by OpenPopup above.
                }
                ...
                ImGui::EndChild();
                ImGui::EndPopup(); // Note this does not do anything to the popup open/close state. It just terminates the content declaration.
            }
ed:Resume();
```

But as soon as I zooming out, the title is shifted left ., more and more while zooming out.
![MEFlow_ikB30epEHp](https://user-images.githubusercontent.com/2878768/134675637-366c73e1-5cc4-403b-9044-d5d0cfb3bb13.png)

If I disable the ed:suspend / ed:resume part , the probleme disappear

**Comment by SadE54** [comment]
An I just realized is as soon such a node was spawned in the editor, the text of my menu bar is disappearing :-/

**Comment by thedmd** [comment]
I'm investigating this

**Comment by SadE54** [comment]
sorry , you can close this one. I forgot to say it was solved : it was due , if I remember well to a wrongly position in the code of the suspend/ resume block. it was acting if there was no suspend /resume.

**Comment by thedmd** [comment]
I get to the bottom of what was wrong.

TL;DR: General guideline is to not call `Suspend()/Resume()` inside `Begin{Horizontal|Vertical}()/End{Horizontal|Vertical}()`.

In more details. Scaling and panning is achieved by manually transforming vertices emitted by ImGui. Suspend() force this transformation, which move vertices from canvas space to screen space.
`End{Horizontal|Vertical}()` also manipulate vertices, by moving them into the right position after all spacing is calculated. Since `Suspend()` already transformed vertices to screen space, layout is applied to screen space vertices. This is why texts are moving out of headers. There is not really a way to counteract because both layouts and canvas 'hack' `ImDrawList` data and there is no way for them to know about each other. In the end, this will be limitation.


To show new popup you can try to use `CanvasToScreen` function. These were added some time ago:
```cpp
ImVec2 ScreenToCanvas(const ImVec2& pos);
ImVec2 CanvasToScreen(const ImVec2& pos);
```
You can use it to transform position for the popup to screen space and show it when it is safe to call `Suspend()`.
